### ISSUE TITLE
ScrollView grows infinitely inside StackLayout

### ISSUE THREAD

**Issue by Dreamescaper** [dotnet/maui#5442]
### Description

If you put ScrollView in StackLayout, ScrollView grows infinitely, and therefore you cannot scroll it.

### Steps to Reproduce

Use the following page content:
```xaml
    <VerticalStackLayout>
        <Label Text="Before"/>

        <ScrollView>
            <VerticalStackLayout Spacing="20">
                <Label Text="1"/>
                <Label Text="2"/>
                <Label Text="3"/>
                <Label Text="4"/>
                <Label Text="5"/>
                <Label Text="6"/>
                <Label Text="7"/>
                <Label Text="8"/>
                <Label Text="9"/>
                <Label Text="10"/>
                <Label Text="11"/>
                <Label Text="12"/>
                <Label Text="13"/>
                <Label Text="14"/>
                <Label Text="15"/>
                <Label Text="16"/>
                <Label Text="17"/>
                <Label Text="18"/>
                <Label Text="19"/>
                <Label Text="20"/>
                <Label Text="21"/>
                <Label Text="22"/>
                <Label Text="23"/>
                <Label Text="24"/>
                <Label Text="25"/>
            </VerticalStackLayout>
        </ScrollView>
        
        <Label Text="After"/>        
    </VerticalStackLayout>
```

ER:
ScrollView is scrollable, "After" label is visible.

AR:
![image](https://user-images.githubusercontent.com/17177729/159125595-63c0ca6b-7443-4811-a439-094bf4b6b74e.png)

### Version with bug

Preview 14 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, Windows, I was *not* able test on other platforms

### Affected platform versions

Android 12

### Did you find any workaround?

Use Grid instead.

**Comment by Dreamescaper** [comment]
Probably same underlying issue as https://github.com/dotnet/maui/issues/4107 .

**Comment by XamlTest** [comment]
Verified repro with Visual Studio Enterprise 2022 main 17.2.0 Preview 2.0 [32223.438.main] on windows, Android and iOS. Repro Project: [5442.zip](https://github.com/dotnet/maui/files/8312713/5442.zip)

**Comment by cshung** [comment]
This issue is investigated thoroughly in #4115, in the investigation, we had a discussion with the MAUI team. The discussion is private, but the conclusion of the discussion is that `StackLayout` having an infinite vertical size is the new expected behavior. That is why we closed #4107 as not a bug.

We understand that it is different from `Xamarin.Forms` and there will be documentation on that.
